File: LeetCode/190. Reverse Bits.cpp
// Author : @AdarshGaur

class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t ans = 0;
        for(int i=0, j=31; i<=j; i++, j--){
            if(n & (1 << i)) ans |= (1<<j);
            if(n & (1 << j)) ans |= (1<<i);
        }
        return ans;
    }
};

File: LeetCode/190. Reverse Bits.java
class Solution {
    public int reverseBits(int n) {
        int reversedBits = 0;
        for(int i=0; i<32; i++){
            reversedBits <<= 1;
            reversedBits |= n&1;
            n >>= 1;
        }
        return reversedBits;
    }
}
